portfolio/src/main/webapp/calendarApi.js

```diff
@@ -191,81 +117,82 @@ function listUpcomingEvents() {
                 (newEventExists, existingEvent) => newEventExists ||
                     eventsEqual(newCalendarEvent, existingEvent),
                 /* initialValue= */ false);
-
-            if (!doesEventExist) {
+ 
+            if (!doesEventExist) { 
               updateCalendarEventList(newCalendarEvent);
             }
           });
         }
       });
 }
-
-/**
- * Retrieves the logged in user's Google email and uses that email
- * to embed the user's Google Calendar on the UI.
- * The calendar is displayed in the user's Google Calendar's time zone.
- */
-function showCalendarView(user) {
-  const userEmail = encodeURIComponent(user.getBasicProfile().getEmail());
-
-  // Fetches the user's Calendar's time zone.
-  gapi.client.calendar.settings.get({setting: 'timezone'})
-      .then((responseTimeZone) => {
-        const userCalendarTimeZone =
-            encodeURIComponent(responseTimeZone.result.value);
-        const calendarViewUrl =
-            'https://calendar.google.com/calendar/embed?src=' + userEmail +
-            '&ctz=' + userCalendarTimeZone + '&mode=WEEK';
-
-        // Updates the UI so that calendar view appears.
-        $('#calendar-view').attr('src', calendarViewUrl).removeClass('d-none');
-      });
-}
-
+ 
 /**
  * Onclick function for 'Looks good to me, export" button.
  * Asks the user for Write access to the API scope.
  */
-function addWriteScope() {
-  var googleUser = googleAuth.currentUser.get();
-  googleUser.grant({scope: SCOPE_READ_WRITE})
+function addCalendarWriteScope() {
+  var googleUser = GoogleAuth.currentUser.get();
+  googleUser.grant({scope: SCOPE_CALENDAR_READ_WRITE})
       .then((response) => {
         $('#export-calendar-message').addClass('d-none');
-        addNewEventsToGoogleCalendar();
+        const scheduledTasks = collectAllScheduledTasks();
+        addNewEventsToGoogleCalendar(scheduledTasks);
       })
       .catch(function(error) {
         handleExportAuthError(error);
       })
 }
-
+ 
 /**
  * Adds the scheduled task items back to the user's Google Calendar.
- * TODO(hollyyuqizheng): fill in the rest of the method once
- * results can be returned from the scheduling algorithm.
  */
 function addNewEventsToGoogleCalendar() {
-  const event = {};
-  addOneEventToGoogleCalendar(event);
+  const scheduledTasks = collectAllScheduledTasks();
+  console.log(scheduledTasks);
+  scheduledTasks.forEach((scheduledTask) => {
+    const task = scheduledTask.task;
+
+    const taskStartTimeMilliseconds = Date.parse(scheduledTask.date);
+    const taskStartTime = new Date();
+    // This is x1000 because the functions takes milliseconds
```

Comment:
Ah could we move this line of comment to line 161?

Reply:
Done.